Sprint1/step2

kovalinaM opened the pull request.

kovalinaM committed:
Подготовка структуры проекта

kovalinaM committed:
Верстка app-header

kovalinaM committed:
Вертска компонентов Burger-ingredients и burger-ingredient

Набросана общая разметка компонентов

kovalinaM committed:
fix: Исправить ошибку в передаче данных из app в burger-ingredient

Доработаны компоненты списка ингредиентов и карточки ингредиента в списке

kovalinaM committed:
feat: Фильтр в списке ингредиентов

Добавлен фильтр в компоненте списка ингредиентов. Первая рабочая версия. Но требует улучшения.

kovalinaM committed:
feat: добавить PropTypes для компонента BurgerIngredient

kovalinaM committed:
Добавить выравнивание по центру  для названий ингредиентов

kovalinaM committed:
fix: переписать логику работы табов компоненте списка ингедиентов

Табы работают при скролле и при клике происходит прокрутка к соответствующему блоку

kovalinaM committed:
fix: доработать скролл в компоненте списка ингредиентов

kovalinaM committed:
часть верстки компонента BurgerConstructor

Пушу только потому что ближайшие пару дней дорабатывать буду за другим компом. А так этот компонент еще делать и делать.

kovalinaM committed:
fix: починить отображение counter в BurgerIngredient

kovalinaM committed:
fix: исправить ошибки propType и пропсах

Ошибки в названиях

kovalinaM committed:
fix: доработка стилей скролла

kovalinaM committed:
fix: исправить propTypes

id заменено на _id в proptypes компонентов списка ингредиентов и самого ингредиента

kovalinaM committed:
fix: удалено mt в css списка ингредиентов

дубль отступа убран, он задан в классах элемента

kovalinaM committed:
feat: верстка компонента конструктора бургера

kovalinaM committed:
обновить заголовок и описание проекта

kovalinaM committed:
Update README.md

kovalinaM committed:
feat: Подключиться к api

Список ингредиентов для соответствующего компонента получить по API. Список ингредиентов для конструктора захардкожен

kovalinaM committed:
feat: создание заготовок компонентов для модальных окон с деталями заказа и описанием ингредиентов

kovalinaM committed:
feat: добавить компонент ModalOverlay

Верстка и стили компнента

kovalinaM committed:
feat: сверстать компонент Modal

kovalinaM committed:
feat: сверстать компоненты  IngredinetDetails и OrderDetails

kovalinaM committed:
feat: верстка компонента Описание ингредиента

kovalinaM committed:
fix: исправить ошибку в импорте стилей в компоненте Описания заказа

kovalinaM committed:
feat: добавить логику события нажатия Esc для модального окна

kovalinaM committed:
feat: lдоработать стили компонентов Modal, OrderDetails BurgerIngredient

kovalinaM committed:
feat: добавить propTypes для Modal

kovalinaM committed:
fest: доработать вывод данных от API в IngredientDetails

Заменить вывод захардкожанных данных на данные, полученные от API  с описанием конкретного ингредиента

kovalinaM committed:
fix: добавить отступы после названия категории списка ингредиентов

kovalinaM committed:
fix: исправить ошибку в записи text в BurgerConstructor

kovalinaM committed:
fix: перезалить картинку done без рамки

Правильно выгрузить картинку с макета без рамки

kovalinaM committed:
fix: изменить появление скрола на auto в списке ингредиентов в конструкторе

kovalinaM committed:
feat: добавить propTypes для ModalOverlay

mikebars1995 commented on src/components/app/app.jsx:
```diff
@@ -0,0 +1,68 @@
+    };
+
+    getIngredients();
+  }, []);
```
**Отлично**


Отлично, что используете пустой массив зависимостей `[]`, чтобы `useEffect` сработал 1 раз

mikebars1995 commented on src/components/app/app.jsx:
```diff
@@ -0,0 +1,68 @@
+      <div className={stylesApp.container}>
+        <AppHeader />
+
+        <main className={stylesApp.sections}>
```
**Отлично**

* Отлично, что не забываете про тег `main` для главной части страницы

mikebars1995 commented on src/components/burger-constructor/burger-constructor.jsx:
```diff
@@ -0,0 +1,120 @@
+export default BurgerConstructor;
+
+BurgerConstructor.propTypes = {
+  ingredients: PropTypes.arrayOf(
```
**Нужно исправить**


Тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

mikebars1995 commented on src/components/burger-ingredients/burger-ingredients.jsx:
```diff
@@ -0,0 +1,111 @@
+          <h2 className="text text_type_main-medium mb-6">Булки</h2>
+          <ul className={`${styles.list} mt-6`}>
+            {listBun.map((ingredient, index) => (
+              <li key={index}>
```
**Нужно исправить**


Для атрибута `key` лучше всего использовать уникальный `id` : у ингридиента как раз есть такой только с нижним подчеркиванием `_id`

[Почему индекс не нужно использовать, можно прочитать тут:]( https://medium.com/russian/%D0%B8%D0%BD%D0%B4%D0%B5%D0%BA%D1%81-%D0%B2-%D0%BA%D0%B0%D1%87%D0%B5%D1%81%D1%82%D0%B2%D0%B5-%D0%BA%D0%BB%D1%8E%D1%87%D0%B0-%D0%B0%D0%BD%D1%82%D0%B8-%D0%BF%D0%B0%D1%82%D1%82%D0%B5%D1%80%D0%BD-56bab8ada309)

mikebars1995 commented on src/components/modal/modal.jsx:
```diff
@@ -0,0 +1,43 @@
+    const keyDownEsc = (e) => (e.key === "Escape" && props.onClose());
+    document.addEventListener('keydown', keyDownEsc);
+    return () => {
+      document.removeEventListener('keydown', keyDownEsc);
```
**Отлично**

Правильно удаляется обработчик `Escape`

mikebars1995 commented on src/components/modal/modal.jsx:
```diff
@@ -0,0 +1,43 @@
+Modal.propTypes = {
+  children: PropTypes.element,
+  header: PropTypes.string,
+  onClose: PropTypes.func
```
**Нужно исправить**

Всем элементам **пропсов**, которые являются обязательными, нужно добавлять `isRequired` при типизации с `propTypes`. Это касается самих **массивов** и **объектов**. В самом конце нужно указывать `isRequired`, если массив сам обязателен в `props`. И проверьте, пожалуйста, потом **консоль на наличие ошибок типизации**

mikebars1995 commented on src/components/app/app.jsx:
```diff
@@ -0,0 +1,68 @@
+        }
+        const data = await res.json();
+        setState((prevState) => ({ ...prevState, data, isLoading: false }));
+      } catch (error) {
```
**Отлично**


Отлично, что ловите возможные ошибки в конце запроса к серверу

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на *гитхабе* все комментарии к коду (нужно прокрутить вниз страницу там)


Работа проделана огромная


Что сделано отлично:


* Отлично, что используете пустой массив зависимостей `[]`, чтобы `useEffect` сработал 1 раз
* Отлично, что не забываете про тег `main` для главной части страницы
* Правильно удаляется обработчик `Escape`
* Отлично, что ловите возможные ошибки в конце запроса к серверу
* Хорошая верстка




Что нужно исправить:

* Тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно
* Для атрибута `key` лучше всего использовать уникальный `id` : у ингридиента как раз есть такой только с нижним подчеркиванием `_id`
* Всем элементам **пропсов**, которые являются обязательными, нужно добавлять `isRequired` при типизации с `propTypes`. Это касается самих **массивов** и **объектов**. В самом конце нужно указывать `isRequired`, если массив сам обязателен в `props`. И проверьте, пожалуйста, потом **консоль на наличие ошибок типизации**






Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.		 

kovalinaM committed:
fix: исправить значение атрибута key

Для атрибута key  использовать уникальный id для ингредиента _id

kovalinaM committed:
fix:  добавить isRequired при типизации с propTypes  обязательным пропсам в Modal и ModalOverlay

kovalinaM committed:
fix:   вынести повторяющийся тип ингредеинта в константу ingredientType в отдельный файл

Тип для ингредиента (propTypes) повторяется в некоторых компонентах. Вынести его в константу ingredientType в отдельный файл types.js в папку utils и импортировать туда, где нужно

kovalinaM committed:
fix: исправить стили выравнивания карточек в списке ингредиентов

Немного не ровная сетка карточек в списке ингредиентов была

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
